### PR TITLE
Corrected the type of `AutocompleteItemWithItems` and adding `labelProps`

### LINKED ISSUE #2764: Fixing the type of `AutocompleteOptionGroup`and adding `labelProps`
### Which package has an issue?

@yamada-ui/react

### Package Version

1.5.0

### Description

Fixing the type of `AutocompleteOptionGroup`and adding `labelProps`.

```tsx
            <AutocompleteOptionGroup
              key={i}
              label={label as string}
              {...(props as HTMLUIProps<"ul">)} // Avoid type conversion here.
            >
              {items.map(({ label, value, ...props }, i) => (
                <AutocompleteOption key={i} value={value} {...props}>
                  {label}
                </AutocompleteOption>
              ))}
            </AutocompleteOptionGroup>

```

And allow the style to be applied to the label as well, like this pull request #2760 

### Link to Reproduction

_No response_

### Steps to Reproduce

2. Click on '...'
3. Scroll down to '...'

### Browser

None

### Operating System

- [ ] macOS
- [ ] Windows
- [ ] Linux
- [ ] iPadOS
- [ ] iOS
- [ ] Android

### Do you know how to fix the issue?

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix?

None

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by zonoryo17**
Closes #2764

## Description

I corrected the type of AutocompleteItemWithItems and added labelProps.

## Is this a breaking change (Yes/No):

No

### CODE REVIEW

File: packages/components/autocomplete/src/use-autocomplete.tsx [hunk 1]
```diff
@@ -60,9 +60,10 @@ type AutocompleteItemWithValue = AutocompleteBaseItem & {
   value?: string
 }
 
-type AutocompleteItemWithItems = AutocompleteBaseItem & {
-  items?: AutocompleteItemWithValue[]
-}
+type AutocompleteItemWithItems = AutocompleteBaseItem &
+  AutocompleteOptionGroupProps & {
```
Comment (hirotomoyamada) [inline packages/components/autocomplete/src/use-autocomplete.tsx:64]: This fix is no longer needed in #2803. Please remove this modification.

File: .changeset/thick-spiders-exist.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@yamada-ui/autocomplete": patch
+---
+
+Fixing the type of AutocompleteItemWithItems and adding labelProps.
```
Comment (hirotomoyamada) [inline .changeset/thick-spiders-exist.md:5]: Please correct this description as well.

https://github.com/yamada-ui/yamada-ui/pull/2760/files#r1762265693

File: packages/components/autocomplete/src/use-autocomplete.tsx [hunk 3]
```diff
@@ -1422,6 +1419,10 @@ export type UseAutocompleteOptionGroupProps = HTMLUIProps<"ul"> & {
    * The label of the autocomplete option group.
    */
   label: string
+  /**
+   * Props for autocomplete option group element.
+   */
+  labelProps?: HTMLUIProps<"span">
```
Comment (hirotomoyamada) [inline packages/components/autocomplete/src/use-autocomplete.tsx:1425]: Do not define props to customize components here.  Set to the `autocomplete-option-group.tsx` side.

File: packages/components/autocomplete/src/use-autocomplete.tsx [hunk 4]
```diff
@@ -473,11 +474,7 @@ export const useAutocomplete = <T extends string | string[] = string>(
           const { label, items = [], ...props } = item
 
           return (
-            <AutocompleteOptionGroup
-              key={i}
-              label={label as string}
-              {...(props as HTMLUIProps<"ul">)}
-            >
+            <AutocompleteOptionGroup key={i} label={label as string} {...props}>
```
Comment (hirotomoyamada) [inline packages/components/autocomplete/src/use-autocomplete.tsx:477]: This fix is no longer needed in #2803. Please remove this modification.

**Comment by hirotomoyamada** [comment]
@zonoryo17 

Conflicts are occurring. Please correct.

File: packages/components/autocomplete/src/use-autocomplete.tsx (hunk 1 above)
Comment (zonoryo17) [reply packages/components/autocomplete/src/use-autocomplete.tsx]: fixed

File: .changeset/thick-spiders-exist.md (hunk 2 above)
Comment (zonoryo17) [reply .changeset/thick-spiders-exist.md]: fixed

File: packages/components/autocomplete/src/use-autocomplete.tsx (hunk 3 above)
Comment (zonoryo17) [reply packages/components/autocomplete/src/use-autocomplete.tsx]: fixed

File: packages/components/autocomplete/src/use-autocomplete.tsx (hunk 4 above)
Comment (zonoryo17) [reply packages/components/autocomplete/src/use-autocomplete.tsx]: fixed

**Comment by zonoryo17** [comment]
@hirotomoyamada 

I have made the correction and resolved the conflicts.
Please review it again.

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**